### PR TITLE
Some additional tests for competition

### PR DESCRIPTION
**Pull request by sedyh**
I've added the tests for sedyh/mizu@experimental and some additional fixes.
From this discussion: #200.

### CODE REVIEW

**Comment by mlange-42** [comment]
🤔 Any idea how I can approve the CI to run? Or why I can't? No such option anywhere...

**Comment by mlange-42** [comment]
Ok, fixed. Could you rebase onto `main` please?

**Comment by sedyh** [comment]
> Ok, fixed. Could you rebase onto `main` please?

Ok, I'll do it tomorrow. It's 4 AM rn at my tz.

**Comment by sedyh** [comment]
Ok. Looks like I've synced a fork and rebased my branch onto main.

**Comment by mlange-42** [comment]
GLFW dependencies are missing. The need to be installed before the benchmarks run. See here for an example: https://github.com/mlange-42/arche-pixel/blob/main/.github/workflows/tests.yml#L24

**Comment by sedyh** [comment]
Is it about changing the build stage?
Is that okay?

**Comment by mlange-42** [comment]
No, it is required for the failing jobs in `benchmarks.yml`.

**Comment by sedyh** [comment]
Ah, I get it. The problem is that I wrapped ebitengine at myself.
Thats because I wanted to make a simple API, but the engine api doesn't really work with di and "accept interface, return struct" principle.

**Comment by mlange-42** [comment]
Yeah, I would try to avoid such dependencies where possible, esp. if it includes something as heavy as OpenGL.

**Comment by mlange-42** [comment]
Mh, another error around GLFW. But this time, it should not happen when just importing it. It looks like Ebiten actually does some GLFW stuff. I got this error only when actually trying to create a window, or something else that requires a "display".

So I guess this won't run in the CI in it's current form.

**Comment by mlange-42** [comment]
@sedyh Finally, I found a way to circumvent the problem of no display in the CI runners.

There is `xvfb`, which emulates a display. Here is an example how to use it:

```yml
  test:
    name: Run tests
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
      - name: Setup Go
        uses: actions/setup-go@v3
        with:
          go-version: '1.20.x'
      - name: Install dependencies
        run: |
          sudo apt-get update -y
          sudo apt-get install -y libgl1-mesa-dev xorg-dev xvfb
          go get .
      - name: Run tests
        run: |
          xvfb-run -a go test -v ./... -covermode=count -coverprofile="coverage.out"
          go tool cover -func="coverage.out"
```

**Comment by mlange-42** [comment]
> Yeah... if we are going to merge that, then, probably, only in a separate branch, so as not to litter with additional dependencies? Or it will just better to close the PR after getting the test results?

I am a bit undecided. Maybe I will merge it, as it adds the dependencies only to the `benchmark` module, which is separate from `arche`.

But maybe I will treat it like the Entitas benchmarks, and merge both branches together. Would then rebase when re-running due to improvements here or in the other implementations.

Will prepare and upload the plots in the next days.

File: benchmark/competition/add_remove/mizu_test.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+func BenchmarkIterMizu(b *testing.B) {
+	b.StopTimer()
+
+	_ = engine.NewGame(func(w engine.World) {
+		position := &Position{}
+		velocity := &Velocity{}
+
+		for i := 0; i < nEntities; i++ {
+			w.AddEntity(&position)
```
Comment (mlange-42) [inline benchmark/competition/add_remove/mizu_test.go:16]: This should not be a pointer (`position` is already a pointer). The effect is that everything below does nothing. Try it by counting `w.Each` loops iterations.

**Comment by mlange-42** [comment]
Another thing I notice while preparing the plots is that something in your PR (a dependency?) is slowing down some of the benchmarks, particularly in the `Build` category. Not sure what this can be.

**Comment by sedyh** [comment]
Hello, sorry for the long answer. The situation with the build slowdown quite strange, I'll try my best to solve it later.

File: benchmark/competition/add_remove/mizu_test.go (hunk 1 above)
Comment (sedyh) [reply benchmark/competition/add_remove/mizu_test.go:16]: Ok, I'll fix it.

**Comment by sedyh** [comment]
Well, it's been sooo long until the last message. I think I should close this PR for now until I figure out the problem so that it doesn’t just hang around.

**Comment by sedyh** [comment]
Arche is nice btw ;)

**Comment by mlange-42** [comment]
Thanks! Actually, IMO it would be useful to keep this issue open.

**Pull request closed without merging**